dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/SecurityUtilsService.kt

```diff
@@ -0,0 +1,38 @@
+package org.dataland.datalandbackend.services
+
+import okhttp3.internal.concurrent.TaskRunner.Companion.logger
+import org.dataland.keycloakAdapter.auth.DatalandAuthentication
+import org.springframework.stereotype.Service
+import java.nio.file.AccessDeniedException
+
+/**
+ * Implements utility functions that can be used e.g., in PRE_AUTHORIZE
+ * for several authentication use-cases
+ */
+@Service("SecurityUtilsService")
```

Comment:
Short call and small rework. 
Summary: that a user can always that for their data ownership and get a 404 error if they are not owner of company. If the head endpoint is used unauthorized a 403 error should be given.

Reply:
ok